dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -32,143 +23,92 @@ namespace PushNotificationHelpers
-            }
-
+            THROW_IF_FAILED(PushNotifications_CloseChannel(m_channelInfo.appId.c_str(), m_channelInfo.channelId.c_str()));
             PushNotificationTelemetry::ChannelClosedByApi(S_OK);
         }
 
         catch (...)
         {
-            auto channelCloseException = hresult_error(to_hresult());
+            auto channelCloseException { hresult_error(to_hresult()) };
 
             PushNotificationTelemetry::ChannelClosedByApi(channelCloseException.code());
 
             if (channelCloseException.code() != HRESULT_FROM_WIN32(ERROR_NOT_FOUND))
             {
-                throw hresult_error(to_hresult());
+                throw channelCloseException;
             }
         }
     }
 
-    winrt::event_token PushNotificationChannel::PushReceived(winrt::Windows::TypedEventHandler<winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel, winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> handler)
+    winrt::event_token PushNotificationChannel::PushReceived(TypedEventHandler<winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel, winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> handler)
     {
-        if (PushNotificationHelpers::IsPackagedAppScenario())
+        bool registeredEvent { false };
         {
-            if (m_channel)
-            {
-                return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
-                {
-                    if (auto strong = weak_self.get())
-                    {
-                        auto pushArgs = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args);
-                        pushArgs.Handled(true);
-                        handler(*strong, pushArgs);
-                    };
-                });
-            }
-            else
-            {
-                // The channelUri is directly obtained when we request Channel from UDK using RemoteId
-                auto lock = m_lock.lock_exclusive();
-
-                if (!m_foregroundHandlerCount)
-                {
-                    auto appUserModelId{ winrt::Microsoft::Helpers::GetAppUserModelId() };
-
-                    THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appUserModelId.get(), this));
-                }
+            auto lock { m_lock.lock_shared() };
+            registeredEvent = bool(m_foregroundHandlers);
+        }
 
-                ++m_foregroundHandlerCount;
+        if(!registeredEvent)
+        {
+            if(PushNotificationHelpers::IsPackagedAppScenario())
+            {
+                auto appUserModelId{ PushNotificationHelpers::GetAppUserModelId() };
 
-                return m_foregroundHandlers.add(handler);
+                // Register foreground event through the UDK
```

Comment:
Not a particularly useful comment. Instead, this would be more descriptive:
"Register a Sink with the platform initialized in the current process"